src/layout/container.styles.tsx

```diff
@@ -0,0 +1,129 @@
+import styled, { css } from "styled-components";
+import { ContainerType } from "./types";
+import { MediaQuery } from "../theme/mediaquery/mediaquery-helper";
+import { Breakpoint } from "../theme";
+
+interface StyleProps {
+    $type: ContainerType;
+    $stretch: boolean;
+}
+
+export const StyledContainer = styled.div<StyleProps>`
+    flex-grow: 1;
+    margin: 0 auto;
+    position: relative;
+    width: auto;
+    height: auto;
+
+    ${(props) => {
+        if (props.$stretch) {
+            return css`
+                padding: 0 32px;
+            `;
+        } else {
+            return css`
+                padding: 0 32px;
+                max-width: 1440px;
+            `;
+        }
+    }}
+
+    ${MediaQuery.MaxWidth.xl} {
+        padding: 0 ${Breakpoint["xl-margin"]};
+    }
+
+    ${MediaQuery.MaxWidth.lg} {
+        padding: 0 ${Breakpoint["lg-margin"]};
+    }
+
+    ${MediaQuery.MaxWidth.md} {
+        padding: 0 ${Breakpoint["md-margin"]};
+    }
+
+    ${MediaQuery.MaxWidth.sm} {
+        padding: 0 ${Breakpoint["sm-margin"]};
+    }
+
+    ${MediaQuery.MaxWidth.xs} {
+        padding: 0 ${Breakpoint["xs-margin"]};
+    }
+
+    ${MediaQuery.MaxWidth.xxs} {
+        padding: 0 ${Breakpoint["xxs-margin"]};
+    }
```

Comment:
this would take override the styles above. only apply this if `$stretch` is false.